static/js/narrow.js

```diff
@@ -355,14 +355,6 @@ exports.activate = function (raw_operators, opts) {
     $('#search_query').val(Filter.unparse(operators));
     search.update_button_visibility();
 
-    if (!had_message_content && opts.trigger === 'sidebar' && exports.narrowed_by_reply()) {
-        if (exports.narrowed_to_topic()) {
-            compose.start('stream');
-        } else {
-            compose.start('private');
-        }
-    }
-
```

Comment:
So this block earlier in the function might close the compose box:
```
    var had_message_content = compose.has_message_content();                                         
                                                                                                     
    if (!had_message_content) {                                                                      
        compose.cancel();                                                                            
    } else {                                                                                         
        compose_fade.update_message_list();                                                          
    }                                                                                                
```

So the code we're deleting here does 2 things:
* Keep the compose box open (and update the recipient) in the case it was already open and empty and you just switched views.
* Open it if the compose box was not open.

I think we want to keep the first behavior, but not the second, right?

See ec56e946a7bdc280dc36806a06cd9ea32e8e5b33 and 89cbda5001d5fc8a0d1a07c5d9a7601f8ca9ffa4 for some history.